server/monitor/src/main/java/org/apache/accumulo/monitor/util/AccumuloMonitorAppender.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.monitor.util;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.accumulo.core.Constants;
+import org.apache.accumulo.core.client.Instance;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.core.zookeeper.ZooUtil;
+import org.apache.accumulo.fate.zookeeper.ZooCache;
+import org.apache.accumulo.fate.zookeeper.ZooCacheFactory;
+import org.apache.accumulo.server.client.HdfsZooInstance;
+import org.apache.log4j.AsyncAppender;
+import org.apache.log4j.net.SocketAppender;
+import org.apache.zookeeper.data.Stat;
+
+import com.google.common.net.HostAndPort;
+
+public class AccumuloMonitorAppender extends AsyncAppender {
+
+  private final ScheduledExecutorService executorService;
+  private final AtomicBoolean trackerScheduled;
+
+  /**
+   * A Log4j Appender which follows the registered location of the active Accumulo monitor service, and forwards log messages to it
+   */
+  public AccumuloMonitorAppender() {
+    // create the background thread to watch for updates to monitor location
+    trackerScheduled = new AtomicBoolean(false);
+    executorService = Executors.newSingleThreadScheduledExecutor(runnable -> {
+      Thread t = new Thread(runnable, AccumuloMonitorAppender.class.getSimpleName() + " Location Tracker");
+      t.setDaemon(true);
+      return t;
```

Comment:
Maybe. I just don't see much difference from remembering to do `new Daemon` instead of `new Thread` and remembering to do `setDaemon(true)`. The former adds a level of obfuscation that a developer has to track, but otherwise, they seem to be equivalent to me from the perspective of what the developer is responsible for remembering to do. The only way the `Daemon` class is better is if we had an executor service whose API explicitly wanted those instead of `Thread`. It's not the case here, but maybe it's the case elsewhere in Accumulo. I haven't looked.